webhook/resources/volume/validator.go

```diff
@@ -331,50 +332,71 @@
~	return nil
~}
~
~func (v *volumeValidator) validateExpansionSize(oldVolume *longhorn.Volume, newVolume *longhorn.Volume) error {
~	oldSize := oldVolume.Spec.Size
~	newSize := newVolume.Spec.Size
~	if newSize == oldSize {
~		return nil
~	}
~	if newSize < oldSize && !newVolume.Status.ExpansionRequired {
 		return fmt.Errorf("shrinking volume %v size from %v to %v is not supported", newVolume.Name, oldSize, newSize)
 	}
 
+	replicas, err := v.ds.ListVolumeReplicasRO(newVolume.Name)
+	if err != nil {
+		return err
+	}
+
+	for _, replica := range replicas {
+		if datastore.IsBackendStoreDriverV1(replica.Spec.BackendStoreDriver) {
```

Comment:
Suggest a common function for this? the similar code used in replica scheduling.

Reply:
Added a common function. Thanks.